Modules/Algorithms/Shor/Shor's.ipynb

```diff
@@ -0,0 +1,330 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook, we demonstrate Shor's Algorithm factoring the integer 15\n",
+    "\n",
+    "1. [Introduction](#intro)\n",
+    "2. [Controlled Modular Exponentiation](#mod)\n",
+    "3. [Quantum Fourier Transform](#qft)\n",
+    "4. [Circuit Construction](#cc)\n",
+    "5. [Execution](#exec)\n",
+    "6. [Post Processing](#post)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id='intro'></a>\n",
+    "## 1. Introduction\n",
+    "\n",
+    "### Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install numpy\n",
+    "%pip install qiskit\n",
+    "%pip install qiskit_ibm_runtime\n",
+    "%pip install qiskit_aer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import math\n",
+    "from qiskit import QuantumCircuit\n",
+    "from qiskit.circuit.library import CUGate\n",
+    "from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager as gppm\n",
+    "from qiskit_ibm_runtime import QiskitRuntimeService, SamplerV2 as Sampler\n",
+    "from qiskit.visualization import plot_distribution\n",
+    "from qiskit_aer import AerSimulator"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Controlled Modular Exponentiation Operation\n",
+    "As an example, say we want to factor N = 15.\n",
+    " $$N^{r} \\%\\ 15$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Shor's Algorithm\n",
+    "\n",
+    "\n",
+    "#### Step 1. Choose a random integer $a$:\n",
+    "   - Pick \\( a = 2 \\).\n",
+    "   - Compute gcd(a, N) . Since  gcd(2, 15) = 1 , we continue.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Step 2. Find the period $r$ of $a$ mod N:\n",
+    "   - The order $r$ is the smallest integer such that:\n",
+    "     $2^r$  = mod 15\n",
+    "    \n",
+    "   - Quantum period-finding determines $r$ = 4."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Step 3. Check if $r$ is even:\n",
+    "   - Since \\( r = 4 \\) is even, we proceed.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Step 4. Compute potential factors of \\(N\\):\n",
+    "\n",
+    "- Compute: $a^{r/2} $ - 1 and $a^{r/2}$ + 1.\n",
+    "\n",
+    "\n",
+    "- Compute gcd with \\( N = 15 \\):\n",
+    "$$\n",
+    "  p = \\gcd(3, 15) = 3, \\quad q = \\gcd(5, 15) = 5\n",
+    "$$\n",
+    "  \n",
+    "- This successfully factors 15 into 3 * 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Circuit of Shor's Algorithm \n",
+    "<img src=\"shor_images/Picture1.png\">"
```

Comment:
This is not the circuit of Shor's algorithm, this is the circuit of Quantum Phase Estimation!

Your image shows inverse QFT but you use QFT. Either is OK but please adjust to make sure the image lines up with the code.

Additionally, please use an image that shows the steps of QPE.